3d/pcb.scad

```diff
@@ -89,11 +97,14 @@ module pcb() {
         }
     }
 
-
-
     // Sensor pins
     color([0.5, 0.5, 0.5]) {
-        translate([pcb_hole_to_sensor_pin_1_x - pcb_sensor_pin_width/2, pcb_hole_to_sensor_pin_1_y - pcb_sensor_pin_width/2, -sensor_pin_length + pcb_thickness + 0.1]) {
+        pin_extra_length = 0.1;  // pins excess sticking out from the back of the PCB
+        sensor_z_offset = pcb_to_spool - sensor_spool_distance - hall_effect_thickness/2 - 0.1;
+        sensor_pin_length = sensor_z_offset + pcb_thickness + pin_extra_length;
+        assert(sensor_pin_length < hall_effect_pin_length_max, "Warning: design is too thick to fit sensor");
```

Comment:
Oh cool, looks like OpenSCAD finally got bumped to a modern version (2019.05) in Ubuntu 20.04 so we can use `assert` instead of `legacyAssert` now; nice!